packages/remote-config/android/src/main/java/io/invertase/firebase/config/UniversalFirebaseConfigModule.java

```diff
@@ -143,9 +155,53 @@ Task<FirebaseRemoteConfigInfo> ensureInitialized(String appName) {
     return ensureInitializedTask;
   }
 
+  ConfigUpdateListenerRegistration onConfigUpdated(String appName, Callback callback) {
+    FirebaseApp firebaseApp = FirebaseApp.getInstance(appName);
+
+    return FirebaseRemoteConfig.getInstance(firebaseApp).addOnConfigUpdateListener(new ConfigUpdateListener() {
+      @Override
+      public void onUpdate(ConfigUpdate configUpdate) {
+        Set<String> updatedKeys = configUpdate.getUpdatedKeys();
+        List<String> updatedKeysList = new ArrayList<>(updatedKeys);
+
+        Map<String, Object> results = new HashMap<>();
+        results.put("updatedKeys", updatedKeysList);
+        callback.invoke(Arguments.makeNativeMap(results), null);
+      }
+
+      @Override
+      public void onError(FirebaseRemoteConfigException error) {
+        WritableMap userInfoMap = Arguments.createMap();
+
+        if (error == null) {
+          userInfoMap.putString("code", "unknown");
+          userInfoMap.putString(
+            "message",
+            "Operation cannot be completed successfully, due to an unknown error.");
+        } else {
+          userInfoMap.putString("nativeErrorMessage", error.getMessage());
+        }
+
+        if (error.getCause() instanceof FirebaseRemoteConfigFetchThrottledException) {
+          userInfoMap.putString("code", "throttled");
+          userInfoMap.putString(
+            "message",
+            "fetch() operation cannot be completed successfully, due to throttling.");
+        } else {
+          userInfoMap.putString("code", "failure");
+          userInfoMap.putString(
+            "message",
+            "fetch() operation cannot be completed successfully.");
+        }
```

Comment:
I think remove all this and propagate error as it is. Here are the docs for `FirebaseRemoteConfigException`. https://firebase.google.com/docs/reference/kotlin/com/google/firebase/remoteconfig/FirebaseRemoteConfigException

You need to use `@NonNull` annotations on this parameter like this:
```
public void onError(@NonNull FirebaseRemoteConfigException error) {
```
Are you using android studio for android dev? It will flag when you need to use it.